src/twisted/python/failure.py

```diff
@@ -252,6 +262,7 @@ def __init__(self, exc_value=None, exc_type=None, exc_tb=None,
                         tb.tb_lineno, (), ()
                         ))
                 tb = tb.tb_next
+            # merging current stack with stack stored in the Failure.
```

Comment:
I don't know if this comment is really needed... same for the above comments.

I prefer to see as little comments as possible. For me, the current code is good enough.

In 6 month we might do a change to the code and forget to update the comment ... and then the comment will just confuse rather than help.

We have the docstring for tests to explain all the use cases which this code should support

Reply:
I added those comments because I did actually write the code 6 month ago, and I couldn't make sense of it at first glance, I needed a good 5 min in order to understand again why I did it like that.
Merging current stack with failure contained stack is the core idea of this trick. This is why I insist on it. 
I think future change for this code needs to carefully understand it.
Please let me know which line exactly you think is superfluous.